### DISCUSSION TITLE
How to include files outside project root from a common folder?

### DISCUSSION THREAD

**Discussion by 1mike12** [infinitered/ignite#1835] [Q&A] [answered]
I have a repo with this structure:

```
-common\
---common.json
-react-native-app
-server
```

I'm trying to import the shared json file into RN but the metro bundler can't include files outside the project from what I've read. Apparently I am supposed to put a `metro.config.js` file under my react-native-app folder to turn on more folders for it to watch but I cannot get the project to pick up on the existence of this file. When I try to just console.log something in the script, it doesn't work. 

```js
//this is for allow importing of common folder outside root
const path = require('path');
const COMMON_PATH = path.resolve(path.join(__dirname, '../common'));
const extraNodeModules = {
    'common': COMMON_PATH,
};
const watchFolders = [
    COMMON_PATH
];
/////////////////////////////

module.exports = {
    transformer: {
        getTransformOptions: async () => ({
            transform: {
                experimentalImportSupport: false,
                inlineRequires: true,
            },
        }),
    },
    resolver: {
        extraNodeModules
    },
    watchFolders,
};

```

**Comment by jamonholmgren** [comment, marked as answer]
I've had nothing but problems trying to do this, myself. If someone "solves" it, that would be epic.

You might even hardlink the folder into your app. That might work.

```
ln ../some/source/folder ./app/external
```

**Reply by jamonholmgren** [reply]
You might also look at rnx-kit, which has an increasing number of utilities to help with Metro configs.

https://github.com/microsoft/rnx-kit#Bundling
